config/backend-config/namespace_config.go

```diff
@@ -10,12 +10,26 @@ import (
 	"time"
 
 	"github.com/cenkalti/backoff"
+	jsoniter "github.com/json-iterator/go"
 
 	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/services/controlplane/features"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 	"github.com/rudderlabs/rudder-server/utils/types"
 )
 
+var jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+
+func init() {
```

Comment:
the package is `backendconfig`. Won't this be called as soon as the `backendconfig` package is imported, regardless if `namespaceConfig` is used or not?

Based on the above, will package init be enough for being able to properly register features during server startup, before the client tries to send them to the control plane?

Reply:
**Question:** after reading the document, what I understood is that the component might be any of the following
- `server`
- `sources`
- `transformer`
- `warehouse`

According to that, I wouldn't expect to see here any component other than `server` or `warehouse`. Why do we register a `backend-config` component?

Reply:
This is something I want to get some feedback on from the control plane team as well. 

> Won't this be called as soon as the backendconfig package is imported, regardless if namespaceConfig is used or not?

Yes, it will. I mention some alternatives approach that will increase the complexity. My understanding so far is that for the first use cases, we just want to know if a feature is implemented in the current rudder-server version. Not if it is enabled or not.

I will ask for feedback again and revisit my approach if needed.

> Why do we register a backend-config component?

Regarding `backend-config, namespace`, I added it as an example, and It will not be used by the control plane. I think it is better to change this into something more useful.

Reply:
>My understanding so far is that for the first use cases, we just want to know if a feature is implemented in the current rudder-server version. Not if it is enabled or not.

In such a case, a statically initialized map would be sufficient and we don't really need a registry

```
features := map[string][]string {
	"server": ["featureA", "featureB"],
	"warehouse": ["featureC", "featureD"],
}
```

Reply:
You are right, as an initial approach this seems to be the most sensible way.